### PR TITLE
morebits: lookupCreation: add failure callback

### PR DESCRIPTION
**Pull request by siddharthvp**
This is unlikely to fail much due to API errors being just a read operation, but things like connection drops can always happen.

There ought to be _some_ way in which a failure can be detected.

### CODE REVIEW

**Review by Amorymeltzer — COMMENTED** [review]
Seems reasonable and good!  Untested atm but 👍

File: morebits.js [hunk 1]
```diff
@@ -3221,12 +3222,13 @@ Morebits.wiki.page = function(pageName, status) {
 	 * @param {Function} onSuccess - Callback function to be called when
 	 * the username and timestamp are found within the callback.
 	 */
```
Comment (Amorymeltzer) [inline morebits.js:3224]:
```suggestion
	* @param {Function} onSuccess - Callback function to be called when the lookup fails
	*/
```

File: morebits.js [hunk 2]
```diff
@@ -3221,12 +3222,13 @@ Morebits.wiki.page = function(pageName, status) {
...
-	this.lookupCreation = function(onSuccess) {
+	this.lookupCreation = function(onSuccess, onFailure) {
 		if (!onSuccess) {
 			ctx.statusElement.error('Internal error: no onSuccess callback provided to lookupCreation()!');
 			return;
```
Comment (Amorymeltzer) [inline morebits.js:3228]:
```suggestion
			ctx.onLookupCreationFailure(this);
			return;
```

File: morebits.js [hunk 3]
```diff
@@ -3221,12 +3222,13 @@ Morebits.wiki.page = function(pageName, status) {
...
 		}
 		ctx.onLookupCreationSuccess = onSuccess;
+		ctx.onLookupCreationFailure = onFailure || emptyFunction;
```
Comment (Amorymeltzer) [inline morebits.js:3231]: Should probably move these up to accomodate returning the failure if `onSuccess` not provided.

File: morebits.js [hunk 4]
```diff
@@ -3848,6 +3850,7 @@ Morebits.wiki.page = function(pageName, status) {
 		var rev = response.pages[0].revisions && response.pages[0].revisions[0];
```
Comment (Amorymeltzer) [inline morebits.js:3850]: Shouldn't we provide the failure function to `fnCheckPageName` on line 3846?

**Review by Amorymeltzer — CHANGES_REQUESTED** [review]
Couple more places for the failure callback, nothing major

File: morebits.js (hunk 4 above)
Comment (siddharthvp) [reply morebits.js:3853]: things get hazy when there are that many lines ....

**Review by Amorymeltzer — APPROVED** [review]

**Pull request merged**